src/main.rs

```diff
@@ -373,7 +373,7 @@ async fn async_run_ssh(opts: SshCommand, params: SshParameters) -> anyhow::Resul
     domain.attach().await?;
 
     let window_id = mux.new_empty_window();
-    let tab = domain.spawn(PtySize::default(), cmd, window_id)?;
+    let tab = domain.spawn(PtySize::default(), cmd, None, window_id)?;
```

Comment:
It would be nice to add a config option to specify the default cwd: on the surface of things it would simply be:

```
    /// or much better name
   default_working_dir:  Option<PathBuf>,
```

but that should probably be domain specific rather than global.
Totally fine to punt on this.

Reply:
When would this default be used? I see one of:
- override tab cwd
- override $HOME
- used by default if one or both of the above don't exist/isn't accessible

Reply:
Heh, and this is where the complexity comes in :)

My initial thought was that the config default would be used:
* when spawning the initial tab in a window (I think all of the call sites in `main.rs` are these)
* For the `SpawnTabDomain::Domain` and `SpawnTabDomain::DomainName` cases where we can't pull the cwd from an existing tab
* Where the tab cwd `is_none`

I can see that someone might potentially want to choose how to resolve the cwd if we have one from the current tab and one in the config file, so perhaps there should be an option to specify whether the cwd from the tab should take precedence.

Reply:
Does it make sense for the default be a property of a domain?

Reply:
Yeah, because a domain can be remote and may have a different filesystem layout.   So I think we'd end up with a "global" default that applies to the default domain (in the `Config` struct), and then the explicitly defined domains would have that value too (`TlsDomainClient`, `UnixDomain`, but not `SshDomain`).

It's probably not very convenient to plumb that config through, so you can feel empowered to punt on this unless you really want to :-)

Reply:
Think I'll pass on it for now.

If you can specify the command, then you can effectively override the current dir anyway, so maybe it isn't all that necessary to have specific config for it. This mechanism is useful when the current dir is a dynamic property of the session.